### PR TITLE
Re-raise the exception that caused LiveTV stream to not open

### PR DESCRIPTION
**Pull request by JustAMan**
**Changes**

Make opening a LiveTV stream re-raise the exception if opening it failed.

**Issues**

Hopefully fixes https://github.com/jellyfin/jellyfin/issues/3065.

**Notes**
Requires testing as I cannot reproduce the issue myself.

cc @gled-rs

### CODE REVIEW

**Comment by gled-rs** [comment]
confirmed as working with the second commit. the backend don't hang anymore on a dead feed.

**Comment by JustAMan** [comment]
So it leaves it open for jf-web part to fix ever-spinning circle... but a simple page refresh as a workaround exists, and spinning circle is a separate bug.

File: Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs [hunk 1]
```diff
@@ -139,13 +152,17 @@ private Task StartStreaming(HttpResponseInfo response, TaskCompletionSource<bool
...
+                    Logger.LogInformation("Copying of {0} to {1} was canceled", GetType().Name, TempFilePath);
+                    openTaskCompletionSource.TrySetException(ex);
                 }
                 catch (Exception ex)
                 {
-                    Logger.LogError(ex, "Error copying live stream.");
+                    Logger.LogError(ex, "Error copying live stream {0} to {1}.", GetType().Name, TempFilePath);
+                    openTaskCompletionSource.TrySetException(ex);
                 }
+                openTaskCompletionSource.TrySetResult(false);
```
Comment (dkanada) [inline Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs:165]:
```suggestion

                openTaskCompletionSource.TrySetResult(false);
```

File: Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs [hunk 2]
```diff
@@ -118,6 +119,18 @@ public override async Task Open(CancellationToken openCancellationToken)
...
+            {
+                // Error happened during opening the stream, re-raise the exception to inform the caller
+                throw taskCompletionSource.Task.Exception;
+            }
+            if (!taskCompletionSource.Task.Result)
+            {
+                Logger.LogWarning("Zero bytes copied from stream {0} to {1} but no exception raised", GetType().Name, TempFilePath);
+                throw new EndOfStreamException(String.Format(CultureInfo.InvariantCulture,
+                                                             "Zero bytes copied from stream {0}",
+                                                             GetType().Name));
```
Comment (dkanada) [inline Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs:132]:
```suggestion
                throw new EndOfStreamException(String.Format(CultureInfo.InvariantCulture, "Zero bytes copied from stream {0}", GetType().Name));
```

File: Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs [hunk 3]
```diff
@@ -118,6 +119,18 @@ public override async Task Open(CancellationToken openCancellationToken)
             //OpenedMediaSource.SupportsDirectStream = true;
             //OpenedMediaSource.SupportsTranscoding = true;
             await taskCompletionSource.Task.ConfigureAwait(false);
+            if (taskCompletionSource.Task.Exception != null)
+            {
+                // Error happened during opening the stream, re-raise the exception to inform the caller
+                throw taskCompletionSource.Task.Exception;
+            }
+            if (!taskCompletionSource.Task.Result)
```
Comment (dkanada) [inline Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs:127]:
```suggestion

            if (!taskCompletionSource.Task.Result)
```

File: Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs [hunk 4]
```diff
@@ -118,6 +119,18 @@ public override async Task Open(CancellationToken openCancellationToken)
             //OpenedMediaSource.SupportsDirectStream = true;
             //OpenedMediaSource.SupportsTranscoding = true;
             await taskCompletionSource.Task.ConfigureAwait(false);
+            if (taskCompletionSource.Task.Exception != null)
+            {
+                // Error happened during opening the stream, re-raise the exception to inform the caller
```
Comment (dkanada) [inline Emby.Server.Implementations/LiveTv/TunerHosts/SharedHttpStream.cs:124]:
```suggestion
                // Error happened while opening the stream so raise the exception again to inform the caller
```

**Pull request merged**